PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/runner/webapp/webappconfig/ui/RiderWebAppSettingPanel.kt

```diff
@@ -0,0 +1,740 @@
+        model.publishableProject = publishableProject
+
+        if (rdoUseExist.isSelected) {
+            model.isCreatingWebApp = false
+            model.subscriptionId = selectedWebApp?.subscriptionId
+            model.webAppId = selectedWebApp?.resource?.id()
+        } else if (rdoCreateNew.isSelected) {
+            model.isCreatingWebApp = true
+
+            model.webAppName = txtWebAppName.text
+            model.subscriptionId = lastSelectedSubscriptionId
+
+            // Resource group
+            if (rdoCreateResGrp.isSelected) {
+                model.isCreatingResourceGroup = true
+                model.resourceGroupName = txtNewResGrp.text
+            } else {
+                model.isCreatingResourceGroup = false
+                model.resourceGroupName = lastSelectedResourceGroupName
+            }
+
+            // App service plan
+            if (rdoCreateAppServicePlan.isSelected) {
+                model.isCreatingAppServicePlan = true
+                model.appServicePlanName = txtCreateAppServicePlan.text
+
+                model.region = lastSelectedLocation
+                model.pricingTier = lastSelectedPriceTier
+            } else {
+                model.isCreatingAppServicePlan = false
+                val appServicePlan = cbExistAppServicePlan.getItemAt(cbExistAppServicePlan.selectedIndex)
+                if (appServicePlan != null) {
+                    model.appServicePlanId = appServicePlan.id()
+                }
+            }
+        }
+    }
+
+    //endregion Read From Config
+
+    //region MVP View
+
+    /**
+     * Render table with existing Web apps to publish
+     *
+     * @param webAppLists - list of web apps to render
+     */
+    override fun renderWebAppsTable(webAppLists: List<ResourceEx<WebApp>>) {
+        btnRefresh.isEnabled = true
+        table.emptyText.text = TABLE_EMPTY_MESSAGE
+
+        val sortedList = webAppLists.sortedWith(Comparator { left, right ->
+            left.subscriptionId.compareTo(right.subscriptionId, ignoreCase = true)
+        })
+
+        cachedWebAppList = sortedList
+        if (sortedList.isNotEmpty()) {
+            val model = table.model as DefaultTableModel
+            model.dataVector.clear()
+
+            var subscriptions: List<Subscription> = ArrayList()
+            try {
+                subscriptions = AuthMethodManager.getInstance().azureManager.subscriptions
+            } catch (exception: Exception) {
+                exception.printStackTrace()
+            }
+
+            for (i in sortedList.indices) {
+                val app = sortedList[i].resource
+                val subscriptionName = subscriptions
+                        .filter { it.subscriptionId() == app.manager().subscriptionId() }
+                        .map { it.displayName() }
+                        .firstOrNull() ?: ""
+                model.addRow(arrayOf(app.name(), app.resourceGroupName(), app.region().name(), subscriptionName))
```

Comment:
Consider disabling in this case

Reply:
We cannot disable  the whole table in that case. There is a very specific and almost unreachable case when we got a web app with a subscription that does not already exists. I would rather skip the web app and continue filling the table.